plugins/filter_aws/aws.h

```diff
@@ -36,14 +36,33 @@
 
 #define FLB_FILTER_AWS_IMDS_INSTANCE_ID_PATH              "/latest/meta-data/instance-id/"
 #define FLB_FILTER_AWS_IMDS_AZ_PATH                       "/latest/meta-data/placement/availability-zone/"
+#define FLB_FILTER_AWS_IMDS_INSTANCE_TYPE_PATH            "/latest/meta-data/instance-type/"
+#define FLB_FILTER_AWS_IMDS_PRIVATE_IP_PATH               "/latest/meta-data/local-ipv4/"
+#define FLB_FILTER_AWS_IMDS_VPC_ID_PATH_PREFIX            "/latest/meta-data/network/interfaces/macs/"
+#define FLB_FILTER_AWS_IMDS_AMI_ID_PATH                   "/latest/meta-data/ami-id/"
+#define FLB_FILTER_AWS_IMDS_ACCOUNT_ID_PATH               "/latest/dynamic/instance-identity/document/"
+#define FLB_FILTER_AWS_IMDS_HOSTNAME_PATH                 "/latest/meta-data/hostname/"
+#define FLB_FILTER_AWS_IMDS_MAC_PATH                      "/latest/meta-data/mac/"
 
 #define FLB_FILTER_AWS_IMDS_V2_TOKEN_HEADER               "X-aws-ec2-metadata-token"
 #define FLB_FILTER_AWS_IMDS_V2_TOKEN_HEADER_LEN           24
 
 #define FLB_FILTER_AWS_AVAILABILITY_ZONE_KEY              "az"
 #define FLB_FILTER_AWS_AVAILABILITY_ZONE_KEY_LEN          2
-#define FLB_FILTER_AWS_INSTANCE_ID_KEY                    "ec2_instance_id"
-#define FLB_FILTER_AWS_INSTANCE_ID_KEY_LEN                15
+#define FLB_FILTER_AWS_INSTANCE_ID_KEY                    "instance_id"
```

Comment:
We can't change this unfortunately- it would break backwards compatibility for existing users.

The reason I chose `ec2_instance_id` as the key name is that its the same key name as is used by FireLens, which also sets that metadata.

Reply:
Oh that's fair. Changed it back to `ec2_instance_id ` and updated the `FLB_FILTER_AWS_INSTANCE_ID_KEY_LEN`.
I checked the [AWS doc](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_firelens.html) which I think is the FireLens key name you mentioned. Not sure if it will affect the new metadata I added like `instance_type`. Didn't find the related doc for other key names. But just keep it in the same format, changed the `instance_type` to `ec2_instance_type`.

Reply:
Instance ID is the only thing here which overlaps with FireLens metadata, so you're good.